examples/ogbn_train.py

```diff
@@ -0,0 +1,246 @@
+
+    pbar.close()
+    loss = total_loss / len(train_loader)
+    approx_acc = total_correct / split_idx['train'].size(0)
+    return loss, approx_acc
+
+
+@torch.no_grad()
+def test(loader: NeighborLoader, val_steps=None) -> float:
+    model.eval()
+
+    total_correct = total_examples = 0
+    for i, batch in enumerate(loader):
+        if val_steps is not None and i >= val_steps:
+            break
+        batch = batch.to(device)
+        batch_size = batch.num_sampled_nodes[0]
+        out = model(batch.x, batch.edge_index)[:batch_size]
+        pred = out.argmax(dim=-1)
+        y = batch.y[:batch_size].view(-1).to(torch.long)
+
+        total_correct += int((pred == y).sum())
+        total_examples += y.size(0)
+
+    return total_correct / total_examples
+
+
+if args.use_gat:
+    from torch_geometric.nn.models import GAT
+    model = GAT(
+        in_channels=dataset.num_features,
+        hidden_channels=num_hidden_channels,
+        num_layers=num_layers,
+        out_channels=dataset.num_classes,
+        dropout=args.dropout,
+        heads=args.num_heads,
+    )
+else:
+    from torch_geometric.nn.models import GraphSAGE
+    model = GraphSAGE(
+        in_channels=dataset.num_features,
+        hidden_channels=num_hidden_channels,
+        num_layers=num_layers,
+        out_channels=dataset.num_classes,
+        dropout=args.dropout,
+    )
+
+model = model.to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=args.lr,
+                             weight_decay=args.wd)
+
+prep_time = round(time.perf_counter() - wall_clock_start, 2)
+print("Total time before training begins (prep_time)=", prep_time, "seconds")
+print("Training...")
+
+val_accs = []
+times = []
+train_times = []
+inference_times = []
+best_val = 0.
+start = time.perf_counter()
+
+model.reset_parameters()
+optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
+
+for epoch in range(1, num_epochs + 1):
+    train_start = time.time()
+    loss, _ = train(epoch)
+    train_end = time.time()
+    train_times.append(train_end - train_start)
+
+    inference_start = time.time()
+    val_acc = test(val_loader)
+
+    inference_times.append(time.time() - inference_start)
+    val_accs.append(val_acc)
+    print(f'Epoch {epoch:02d}, Loss: {loss:.4f}, ',
+          f'Train Time: {train_end - train_start:.4f}s')
+    print(f'Val: {val_acc * 100.0:.2f}%,')
+
+    if val_acc > best_val:
+        best_val = val_acc
+    times.append(time.time() - train_start)
+
+val_acc = torch.tensor(val_accs)
+print('============================')
+print("Average Epoch Time on training: {:.4f}".format(
+    torch.tensor(train_times).mean()))
+print("Average Epoch Time on inference: {:.4f}".format(
+    torch.tensor(inference_times).mean()))
+print(f"Average Epoch Time: {torch.tensor(times).mean():.4f}")
+print(f"Median time per epoch: {torch.tensor(times).median():.4f}s")
+print(f'Final Validation: {val_acc.mean():.4f} ± {val_acc.std():.4f}')
```

Comment:
It doesn't make much sense to report the mean and std of validation accuracy over epochs. People usually care about the best validation metric, so I'd suggest printing the best validation accuracy instead.